src/java.desktop/macosx/classes/sun/java2d/macos/MacOSFlags.java

```diff
@@ -75,12 +95,83 @@ private static boolean getPropertySet(String p) {
     private static void initJavaFlags() {
         java.security.AccessController.doPrivileged(
                 (PrivilegedAction<Object>) () -> {
-                    metalEnabled = getBooleanProp("sun.java2d.metal", false);
+                    PropertyState oglState = getBooleanProp("sun.java2d.opengl", PropertyState.UNSPECIFIED);
+                    PropertyState metalState = getBooleanProp("sun.java2d.metal", PropertyState.UNSPECIFIED);
+
+                    if (metalState == PropertyState.UNSPECIFIED) {
+                        if (oglState == PropertyState.DISABLED) {
+                            oglEnabled = false;
+                            metalEnabled = true;
+                        } else if (oglState == PropertyState.ENABLED || oglState == PropertyState.UNSPECIFIED) {
```

Comment:
There are only 3 states aren't there ? And you don't have another branch, so why do we need
"if (oglState == PropertyState.ENABLED || oglState == PropertyState.UNSPECIFIED"
One of those has to be true if we reach here, doesn't it ?